R/graph.R

```diff
@@ -53,164 +130,176 @@ create_bigraph <- function(caobj,
                          marker_genes = NULL,
                          method = BiocNeighbors::KmknnParam(),
                          BPPARAM = BiocParallel::SerialParam()){
-    
+
+
     # apply vector augmentation for MIP search via euclidean distance.
     Xt <- add_zero_dim(caobj@std_coords_cols)
     Qt <- augment_vector(caobj@prin_coords_rows)
-    
+
     cgg_nn <- BiocNeighbors::queryKNN(X = Qt,
                        query = Xt,
                        k = k_cg,
                        get.distance = FALSE,
                        BNPARAM = method,
                        BPPARAM = BPPARAM)$index
-    
+
     cgg_nn <- indx_to_spmat(indx_mat = cgg_nn,
                             row_names = rownames(caobj@std_coords_cols),
                             col_names = rownames(caobj@prin_coords_rows))
-    
+
 
     # gene_idx <- seq_len(nrow(caobj@prin_coords_rows))
-    
+
     if (!is.null(marker_genes)){
         stopifnot(is(marker_genes, "character"))
-        
+
         idx <- which(colnames(cgg_nn) %in% marker_genes)
-        
+
         if (length(idx) == 0){
             warning("Marker genes not found in the data.")
             marker_genes <- NULL
-            
+
         } else {
-            
+
             if(length(idx) < length(marker_genes)){
                 warning("Not all marker genes are in the provided data.")
                 marker_genes <- marker_genes[marker_genes %in% colnames(cgg_nn)]
             }
-            
+
+
             marker_knn <- cgg_nn[,idx, drop=FALSE]
             cgg_nn <- cgg_nn[,-idx, drop = FALSE]
-            
+
+
 
         }
-        
+
     }
-    
+
 
     ccg_nn = BiocNeighbors::findKNN(caobj@prin_coords_cols,
                      k=k_c,
                      get.distance = FALSE,
                      BNPARAM=method,
                      BPPARAM = BPPARAM)$index
-    
+
     if (isTRUE(loops)){
         ccg_nn <- cbind(seq_len(nrow(ccg_nn)),
                         ccg_nn[, -ncol(ccg_nn), drop = FALSE])
     }
-    
-    ccg_nn <- indx_to_spmat(indx_mat = ccg_nn, 
-                            row_names = rownames(caobj@prin_coords_cols), 
+
+    ccg_nn <- indx_to_spmat(indx_mat = ccg_nn,
+                            row_names = rownames(caobj@prin_coords_cols),
                             col_names = rownames(caobj@prin_coords_cols))
 
-    
+
     if (isTRUE(select_genes)){
-      
+
       idx <- Matrix::colSums(cgg_nn) > 0
       cgg_nn <- cgg_nn[,idx, drop = FALSE]
 
+
       if (isTRUE(prune_overlap)){
-        
-        overlap_mat <- calc_overlap( cc_adj = ccg_nn,
-                                     cg_adj = cgg_nn)
-        
+
+        cgg_nn <- calc_overlap( cc_adj = ccg_nn,
```

Comment:
See comment for calc_overlap. I think we should not assign it to cgg_nn because calc_overlap does not return cgg_nn but modifies it in memory.